Fix broken links 154982414

dhandy2013 opened the pull request.

dhandy2013 committed:
154982414: work in progress - fix broken reference links

dhandy2013 committed:
154982414: work-in-progress

dhandy2013 committed:
154982414: Report broken, fixable links

dhandy2013 committed:
fix-broken-links-154982414: Improve link reporting

dhandy2013 committed:
fix-broken-links-154982414: Manually add labels and fix broken links

dhandy2013 committed:
fix-broken-links-154982414: More script fixes, save fixable links

dhandy2013 committed:
fix-broken-links-154982414: Fix all the remaining broken links

dhandy2013 committed:
154982414: Update docstring and comments to reflect what script really does

dhandy2013 committed:
154982414: README.rst: how to do cross-referencing

dhandy2013 committed:
154982414: More specific link to Sphinx doc

dhandy2013 committed:
154982414: Link direct to Sphinx :ref: doc

jonkeane commented:
Ok, I've gone through the Feature Guide and the Endpoint Reference, checking every link I found. I'm sure I missed some, but here are the things that I've found so far:


# Feature guide
- [x] feature-guide/feature-authentication.html

> See the public endpoint reference for more on how to authenticate with OAuth.

Links to first header, not full page


- [x] feature-guide/feature-importing.html

> It is possible to create derived variables using any of the derivation functions available simultaneously in 

Links to first header, not full page


- [x] feature-guide/feature-deriving.html

> Returns a categorical variable with its categories following the specified conditions from different variables on the dataset. View Case Statements.

To fit the style of others: "See [Case Statemnets](url)" instead of "View [Case Statemnets](url)"


- [x] feature-guide/feature-analyzing.html

> See the Endpoint Reference for details.

There should be a link here

> Crunch provides a separate catalog where you can define and manage these common sets of variables. Like most catalogs, you can GET it to see which multitables are defined.

The GET link should be to the GET header, not the first header on the page.



# Endpoint Reference

- [x] endpoint-reference/endpoint-account.html

> If the authenticated user is an account manager, the response will include paths to these additional catalogs: * Account projects * Account teams * Account datasets

This looks like a misparsed bulleted list?


- [x] endpoint-reference/endpoint-batch.html

> Later, using the expressions obtained, it is possible to append all the datasets to a new one without conflicts.

The languages specific code block after this one has three duplications

> See Batches for more details on batch entities and conflicts.

Links to first header, not full page


- [x] endpoint-reference/endpoint-dataset.html

> Datasets are the primary containers of statistical data in Crunch. Datasets contain a collection of variables, with which analyses can be composed, saved, and exported.

The link to variables takes you to the same page, which is really just another link later, this should probably be to endpoint-reference/endpoint-variable.html instead.

> Authorizations on this dataset; see Permissions

Links to first head instead of page

> The body must contain a “name”. You can also include a Crunch Table in a “table” key, as discussed in the Feature Guide. The full set of possible attributes to include when POSTing to create a new dataset entity are:

The link on `Feature Guide` is broken and doesn't go to the correct place (I think it should go to: feature-guide/feature-importing.html#metadata-document-csv)

> A Shoji Catalog of datasets that have been shared with this team. These datasets are not included in the primary dataset catalog. See teams for more.

Links to first head instead of page

> A Shoji Catalog of datasets that belong to this project. These datasets are not included in the primary dataset catalog. See projects for more.

Links to first head instead of page

> Returns all the batches (successful and failed) used for this dataset. See Batches.

Links to a section that has more links, should probably just link directly to the batches endpoint page.

> Contains the list of all datasets joined to the current dataset. See Joins.

Links to an empty section — is there a better page to link to?

> Catalog of all public variables of this dataset. See Variables.

Links to a section that has more links, should probably just link directly to the variables endpoint page.

> Lists the saved versions for this dataset. See Versions.

Links to first head instead of page

> Makes available the public and user-created filters. See Filters.

Links to a section that has more links, should probably just link directly to the filters endpoint page.

> Similar to filters, displays all available multitables. See Multitables

Links to first head instead of page

> Contains all available comparisons. See Comparisons.

Links to an empty section — is there a better page to link to?

> Returns the list of all users and teams with access to this dataset. See Permissions.

Links to first head instead of page

> See Multidimensional Analysis.

Links to first head instead of page


- [x] endpoint-reference/endpoint-join.html

> A variables catalog which describes variables in the subordinate dataset. See Variables for more details.

This links to #catalog on the variable page, when it should really be to the full variable page instead.


- [x] endpoint-reference/endpoint-multitable.html

> POST a Shoji Entity to this catalog to create a new multitable definition. Entities must include a name and template; the template must contain a series of objects with a query and optionally `transform <#transforming-analyses-for-presentation>`__. If omitted, is_public defaults to false. In similar fashion, team will default to null unless a specific team URL is provided.

Something when wrong with the link for transforming-analyses-for-presentation here, I think.

> Multitable entities have a “tabbook” view. See below.

There is no below anymore, so should have a link of some sort (probably to ???)  


- [x] endpoint-reference/endpoint-variable.html

> You may instead also include an “derivation” to derive a variable as a function of other variables. In this case, “type” is not required because it depends on the output of the specified derivation function. For details and examples, see Deriving Variables.

Links to first head instead of page

dhandy2013 committed:
154982414: Fixes based on review feedback

Also filled in some empty sections and got rid of remaining non-Sphinx-style
internal links.

dhandy2013 commented:
@jonkeane Thank you so much for your detailed review. I have fixed all of the items in your punch list, plus many more. In particular, I have filled in all the empty sections at the end of endpoint-reference/endpoint-dataset.html and also fixed the last remaining non-Sphinx-style internal cross references (fixing a couple broken links in the process.)

dhandy2013 committed:
154982414: Fill in some more empty sections and missing links.

jonkeane committed:
fixed one malformed link

jonkeane commented:
This looks good to me, I pushed one minor typo change that was making a link not resolve.
